### PR TITLE
Mostly remove the use of the OPENXR ifdef

### PR DESCRIPTION
**Pull request by hrydgard**
That is, build all the VR code regardless of platform. This is a first step towards making it possible to autodetect and use OpenXR dynamically on Windows - we don't want to build a separate exe or something like that.

This required adding a new dynamic OpenXR loader, so that things can link even when we're not using the static loaders provided by the headset makers.

The new dynamic loader is entirely untested but should work - if not I'll fix it when we actually start doing VR on desktop. In any case, it fixes linking.

### CODE REVIEW

File: Common/GPU/Vulkan/VulkanLoader.cpp [hunk 1]
```diff
@@ -309,7 +309,7 @@ void VulkanSetAvailable(bool available) {
 
 bool VulkanMayBeAvailable() {
 	// Unsupported in VR at the moment
-	if (IsVRBuild()) {
+	if (IsVREnabled()) {
 		return false;
 	}
```
Comment (unknownbrackets) [inline Common/GPU/Vulkan/VulkanLoader.cpp:315]: Although fine for now, I think it'll end up making sense to reverse this onto where VR is configured (until VR is available with Vulkan.)

-[Unknown]

File: Common/VR/OpenXRLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
+#include "Common/VR/OpenXRLoader.h"
+#ifdef _WIN32
+#include "Common/CommonWindows.h"
+#endif
+
+#ifdef XR_NO_PROTOTYPES
+
+PFN_xrGetInstanceProcAddr                  xrGetInstanceProcAddr;
```
Comment (unknownbrackets) [inline Common/VR/OpenXRLoader.cpp:8]: nit: Prefer spaces for alignment, h too...?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]

**Review by lvonasek — APPROVED** [review]
I quickly tested this on Pico Neo 3 Link VR headset. It still works fine.

File: Common/GPU/Vulkan/VulkanLoader.cpp (hunk 1 above)
Comment (hrydgard) [reply Common/GPU/Vulkan/VulkanLoader.cpp:315]: Yes, this is just a minimal step, how configuration works will be redone later.

**Pull request merged**